### PR TITLE
[FEATURE] Disable save button until changes are made to the profile through forms #7853

### LINKED ISSUE #7853: [FEATURE] Disable save button until changes are made to the profile through forms
### Description

image
Can be done to save hits to the api till changes are made to the profile

### Screenshots

_No response_

### Additional information

Other forms should also have this feature
- profile form
- link form
- milestone form
- event form

### PR DESCRIPTION
**Pull request by mianakbarjan**
## [Feature] Issue #7853 

Closes #7853 

## Changes proposed
[save button disable when no changes made](https://github.com/EddieHubCommunity/LinkFree/commit/36600b897e000e3c43d3ef52b8889ab50e427669)
[save button in link form disabled if no changes are made](https://github.com/EddieHubCommunity/LinkFree/commit/02ef6a014b7cee7af06cbe081a32971c1120d322)
[save button in milestone form disabled if no changes are made](https://github.com/EddieHubCommunity/LinkFree/commit/5f9ab0b3f850bb6fb2dc38af56ff639403123532)
[save button in event form disabled if no changes are made](https://github.com/EddieHubCommunity/LinkFree/commit/17373564fe93ec284361baf84927973e63728d5d)

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Before of all 4 forms:
Profile form:
image
Link form:
image
Milestone form:
image
Event form:
image

## After of all 4 forms:
Profile form:
image
Link form:
image
Milestone form:
image
Event form:
image

## Note to reviewers
The code for the date formatting in the event form was slightly changed because refreshing the page of an event would make the start and end dates input blank whereas now it shows the correct start and end dates of the events. There was also a typo in the title of the Manage Milestone page which was fixed. 

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/7944"></a>

### CODE REVIEW

**Comment by amandamartin-dev** [comment]
@EmmaDawsonDev if you have time, would love your expertise on accessibility and disabled buttons here. I read some articles, but didn't find a conclusive answer. Want to make sure this is a good practice.

**Comment by EmmaDawsonDev** [comment]
Disabled buttons don't get read out by screen readers so they are basically invisible. I think most accessibility specialists would advise you to just let users submit the form anyway and use client side validation and feedback to the user instead of disabling buttons.

**Comment by mianakbarjan** [comment]
What do I do now? Is this issue going to be edited to cater to the client side validation and feedback or will it be closed and a new issue opened?

File: pages/account/manage/event/[[...data]].js [hunk 1]
```diff
@@ -65,16 +68,32 @@ export default function ManageEvent({ BASE_URL, event }) {
...
+
+  useEffect(() => {
+    const isEventChanged =
+      name !== event.name ||
+      description !== event.description ||
+      url !== event.url ||
+      startDate !== (event.date?.start ? formatDate(event.date?.start) : '') ||
+      endDate !== (event.date?.end ? formatDate(event.date?.end) : '') ||
+      isVirtual !== (event.isVirtual ? true : false) ||
+      price !== (event.price?.startingFrom || 0);
```
Comment (eddiejaoude) [inline pages/account/manage/event/[[...data]].js:87]: This is difficult to read 🤔

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Is there a way to make it more readable and also more manageable - I feel if there are more form fields in the future they will be forgotten to be added to the checks

**Comment by amandamartin-dev** [comment]
> Disabled buttons don't get read out by screen readers so they are basically invisible. I think most accessibility specialists would advise you to just let users submit the form anyway and use client side validation and feedback to the user instead of disabling buttons.

Thank you @EmmaDawsonDev, I had a feeling this was the case.  @eddiejaoude please see Emmas thoughts on accessibility and disabled buttons. Perhaps we should not implement this change in this way

**Comment by eddiejaoude** [comment]
Oh I missed the previous comment 😱 - thanks @amandamartin-dev for highlighting it (and thanks @EmmaDawsonDev for the explanation)

Maybe for now we should close the issue and PR - then have a discussion on the best way to handle this

**Comment by eddiejaoude** [comment]
As there are accessibility concerns...

Thank you @mianakbarjan for your great work 👍  but I think for now we should close this PR and issue, then continue the discussion elsewhere

**Comment by mianakbarjan** [comment]
> As there are accessibility concerns...
> 
> Thank you @mianakbarjan for your great work 👍  but I think for now we should close this PR and issue, then continue the discussion elsewhere 

Thank you, had a great time working on it. Will check the issues and ask for another issue soon!

**Comment by eddiejaoude** [comment]
Sorry again @mianakbarjan for having to close the issue, I really appreciate your hard work and for your understanding

**Pull request closed without merging**